actfw_core/unicam_isp_capture.py

```diff
@@ -31,8 +38,34 @@
 # TODO: support other than imx219
 # pick from https://github.com/kbingham/libcamera/blob/22ffeae04de2e7ce6b2476a35233c790beafb67f/src/ipa/raspberrypi/data/imx219.json#L132-L142 # noqa: E501, B950
 
-V1_UNICAM_SIZES: List[Tuple[int, int]] = [(2592, 1944), (1920, 1080), (1296, 972), (640, 480)]
-V2_UNICAM_SIZES: List[Tuple[int, int]] = [(3280, 2464), (1920, 1080), (1640, 1232), (640, 480)]
+MAX_LS_GRID_SIZE = 0x8000
+LS_TABLE_W = 16
+LS_TABLE_H = 12
+
+
+# correspond to [libcamera's CameraMode](https://github.com/raspberrypi/libcamera/blob/3fad116f89e0d3497567043cbf6d8c49f1c102db/src/ipa/raspberrypi/controller/camera_mode.h#L19) # noqa: E501, B950
+@dataclass(frozen=True, eq=True)
+class CameraMode:
+    size: Tuple[int, int]
+    scale: Tuple[float, float]
+    crop: Tuple[int, int]
+
+
+V1_SENSOR_SIZE = (2592, 1944)
+V1_UNICAM_MODES: List[CameraMode] = [
+    CameraMode(size=(2592, 1944), scale=(1.0, 1.0), crop=(0, 0)),
+    CameraMode(size=(1920, 1080), scale=(1.0, 1.0), crop=(336, 432)),
+    CameraMode(size=(1296, 972), scale=(2.0, 2.0), crop=(0, 0)),
+    CameraMode(size=(640, 480), scale=(2.0, 2.0), crop=(656, 492)),
```

Comment:
Where do crop values come from?

Reply:
There must be a more appropriate source, but I refer to picamera document.
https://picamera.readthedocs.io/en/release-1.13/fov.html#sensor-modes
> Modes with partial FoV capture from the center of the sensor.

Reply:
>  I refer to picamera document.
> https://picamera.readthedocs.io/en/release-1.13/fov.html#sensor-modes

I looked it again and noticed that the parameters for v1 640x480 don't match the table.
https://github.com/Idein/actfw-core/pull/67/commits/09e0a0ae607d394f21bb7c7ec8aaebe9bf03f494